File: C#/Refactoring/refactoring_exercise_2/test/za/co/entelect/refactoring2/controller/StrategyExampleTest.cs
using NUnit.Framework;
using NUnit.Framework.Constraints;

namespace refactoring_exercise_2.test.za.co.entelect.refactoring2.controller
{
    [TestFixture]
    public class StrategyExampleTest {

        private StrategyExample strategyExample = new StrategyExample();

        [Test]
        public void TestCalculates(){
            Assert.That(strategyExample.Calculate(StrategyExample.Add, 10, 2), new EqualConstraint(12));
            Assert.That(strategyExample.Calculate(StrategyExample.Subtract, 10, 2), new EqualConstraint(8));
            Assert.That(strategyExample.Calculate(StrategyExample.Multiply, 10, 2), new EqualConstraint(20));
            Assert.That(strategyExample.Calculate(StrategyExample.Divide, 10, 2), new EqualConstraint(5));
        }
    }
}

File: Java/refactoring_checkpoint1/src/test/java/za/co/entlect/refactoring1/controller/StrategyExampleTest.java
package za.co.entlect.refactoring1.controller;

import org.junit.Test;

import static junit.framework.Assert.assertEquals;

public class StrategyExampleTest {

    private StrategyExample strategyExample = new StrategyExample();

    @Test
    public void testCalculates(){
        assertEquals(12, strategyExample.calculate(StrategyExample.ADD, 10, 2));
        assertEquals(8, strategyExample.calculate(StrategyExample.SUBTRACT, 10, 2));
        assertEquals(20, strategyExample.calculate(StrategyExample.MULTIPLY, 10, 2));
        assertEquals(5, strategyExample.calculate(StrategyExample.DIVIDE, 10, 2));
    }
}